ballerina/types.bal

```diff
@@ -38,41 +39,65 @@ type AccessTokenResponse record {|
 |};
 
 type DashBoard record {
-    string url = "";
+    string url;
     int heartbeatInterval = 10;
     decimal waitTimeForServicesInSeconds = 5;
-    string groupId = "";
-    string nodeId = "";
-    string mgtApiUrl = "";
+    string groupId;
+    string nodeId = uuid:createType4AsString();
+    string mgtApiUrl;
 };
 
 public type Artifacts record {
     int count;
     Artifact[] list;
 };
 
-public type Artifact Service;
+public enum ArtifactType {
+    SERVICE = "services",
+    LISTENER = "listeners"
+}
+
+public type Artifact Service|Listener;
+
+public type ArtifactDetail ServiceDetail|ListenerDetail;
 
 public type Service record {
     string name;
-    string? attachPoint;
-    Metadata metadata;
-    map<anydata> annotations;
+    string? basePath;
 };
 
-public type Metadata record {
+public type ServiceDetail record {
+    string package;
     Listener[] listeners;
-    map<anydata> metadata;
+    Resource[] resources;
+};
+
+public type Resource record {
+    string[] methods;
+    string url;
 };
 
 public type Listener record {
-    string 'type;
+    string name;
     string? protocol;
-    map<anydata> properties;
+    int port;
+};
+
+public type RequestLimit record {
+    int maxUriLength;
+    int maxHeaderSize;
+    int maxEntityBodySize;
+};
+
+public type ListenerDetail record {
+    string package;
+    string httpVersion;
```

Comment:
Shall we move this fields to native side based on the protocol. This type can be open